### PR TITLE
Great Path Refactor of 2022

### PR DESCRIPTION
**Pull request by cedrickcooke**
This ticket heavily refactors `Path` such that end-users of `Kanvas` no longer have to worry about the actual platform implementation of paths.

`Path` now internally caches platform types after constructing them on first use. In general, this should be an efficiency win as a Krayon `Path` retained between multiple draw passes will now automatically avoid re-allocating the native platform type.

# Breaking Changes

- `Kanvas` is no longer generic on `PATH`
    - Removed `buildPath` function
    - All functions accepting generic `PATH` now accept `Path`
- `Clip` is no longer generic on `PATH`. Now accepts `Path`
- `Path` is no longer a `fun interface` and is now a concrete `class`
- It's now harder/more roundabout to create platform specific types from a Krayon `Path`
    - Removed `Path.toCompose`
    - Removed `Path.toAndroid` and `androidPath`
    - Removed `Path.toPathString`
    - Removed `Path.toPath2d`
    - Removed `PathBuilder<P>.build(Path): P`
- Removed `Path.reify`, as all paths are now instantly reified.
- `ToStringFormatter` is now an `object` instead of a `class`

### CODE REVIEW

**Review by twyatt — APPROVED** [review]
Awesome clean up! 🧹 
Great code comments where they were warranted.
Love the removal of a bunch of generics. ❤️ 
Overall, amazing work! 💯

File: kanvas/src/commonMain/kotlin/Path.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.juul.krayon.kanvas
+
+public class Path internal constructor(
+    private val instructions: SegmentedPath,
+) {
+
+    public constructor(
+        path: PathBuilder<*>.() -> Unit
```
Comment (davidtaylor-juul) [inline kanvas/src/commonMain/kotlin/Path.kt:8]: 🚀

File: kanvas/src/commonMain/kotlin/PathTypeMarker.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+/**
+ * Used to bridge the gap between a [Path] and the platform type required by a [Kanvas].
+ *
+ * Ultimately, [Path] uses instances of this class as a [Map] key, so instances _must_ correctly
+ * implement [hashCode]. This is automatic when implementations take the form of an `object` instead
+ * of a `class`.
+ */
+public interface PathTypeMarker<P : Any> {
+    public val builder: PathBuilder<P>
+}
```
Comment (davidtaylor-juul) [inline kanvas/src/commonMain/kotlin/PathTypeMarker.kt:12]: This is cool. I'm stealing this pattern.

**Review by davidtaylor-juul — APPROVED** [review]
Wicked.

**Pull request merged**